tiledb/sm/array_schema/domain.cc

```diff
@@ -223,19 +223,22 @@ int Domain::cell_order_cmp_impl<char>(
   assert(dim->var_size());
   (void)dim;
 
-  auto var_a{a.value_as<const char[]>()};
-  auto var_b{b.value_as<const char[]>()};
+  auto var_a = reinterpret_cast<const char*>(a.content());
+  auto var_b = reinterpret_cast<const char*>(b.content());
```

Comment:
Why not change `value_as` to use `reinterpret_cast`? This seems a small step backward in type safety.